src/provider/provider.py

```diff
@@ -0,0 +1,49 @@
+import typing
+from abc import abstractmethod
+from numbers import Number
+
+
+class AbstractProvider:
+    @abstractmethod
+    def get_name(self) -> str:
```

Comment:
This could probably just be an attribute on the class itself, right? `name = None`. Maybe this is better though.